[directxtex] Use CMake build system, add OpenEXR support and dx12 feature

RT2Code opened the pull request:
Use the provided CMake file instead of the VS solutions to build the library, as recommended in the maintainer guidelines. DirectXTex can be built without DirectX12 (default for desktop) for Windows 7 Service Pack 1 or later, or with DirectX12 (default for ARM or UWP) for Windows 10.

Also add the openexr feature to enable OpenEXR format support in the library and its tools : https://github.com/Microsoft/DirectXTex/wiki/Adding-OpenEXR

RT2Code committed:
[directxtex] Use CMake build system and add OpenEXR support

Use the provided CMake file instead of the VS solutions to build the library, as recommended in the vcpkg maintainer guide.

Also add the openexr feature to enable OpenEXR in the library and its tools.
https://github.com/Microsoft/DirectXTex/wiki/Adding-OpenEXR

RT2Code committed:
Merge branch 'master' into directxtex

RT222 commented:
Regressions indeed, otherwise it would have been too easy. :) I'll try to fix that as soon as possible.

RT2Code committed:
[directxtex] Fix build failure on ARM and UWP

Hoikas commented on ports/directxtex/enable_openexr_support.patch:
```diff
@@ -0,0 +1,162 @@
++//     private:
++//         HANDLE m_handle;
++//     };
++// }
```
The patch could be smaller if you use `#if 0`

RT222 replied:
You're right, but I did that because I wanted to follow the instructions very closely and they ask to comment this section out. Do you really think I should change it? I have to admit that I fail to see the importance of gaining a few lines on a patch.

Hoikas replied:
I think the goal with commenting sections out is to use block-style comments. While we could use `/* */`, those could easily be broken by upstream changes. If the size of the patch is decreased, we reduce the chance that trivial upstream changes will cause the patch to fail when the port is upgraded.

RT222 replied:
Good point, I'm changing it.

Hoikas commented on ports/directxtex/portfile.cmake:
```diff
@@ -8,63 +8,57 @@ vcpkg_from_github(
+file(INSTALL ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/bin/CMake/Debug/DirectXTex.lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
+file(INSTALL ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/bin/CMake/Release/DirectXTex.lib DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
+if(NOT VCPKG_TARGET_IS_UWP AND NOT TRIPLET_SYSTEM_ARCH STREQUAL "arm64")
+    file(INSTALL ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/bin/CMake/Release/texassemble.exe DESTINATION ${CURRENT_PACKAGES_DIR}/tools/${PORT})
+    file(INSTALL ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/bin/CMake/Release/texconv.exe DESTINATION ${CURRENT_PACKAGES_DIR}/tools/${PORT})
+    file(INSTALL ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/bin/CMake/Release/texdiag.exe DESTINATION ${CURRENT_PACKAGES_DIR}/tools/${PORT})
```
```suggestion
    vcpkg_copy_tools(
        TOOL_NAMES texassemble texconv texdiag
        SEARCH_DIR ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/bin/CMake/Release
    )
```

RT222 replied:
I didn't know about vcpkg_copy_tools, thanks.

Hoikas commented on ports/directxtex/portfile.cmake:
```diff
@@ -8,63 +8,57 @@ vcpkg_from_github(
 endif()
 
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Trailing newline please :smile:

RT222 replied:
Whoopsie. :D

RT2Code committed:
[directxtex] Use vcpkg_copy_tools

RT2Code committed:
[directxtex] Add win7 feature

DirectXTex can be build without DirectX12 to support Windows 7 Service Pack 1 or later platforms, so I added this as a feature.

Hoikas commented on ports/directxtex/portfile.cmake:
```diff
@@ -30,15 +30,16 @@ endif()
     OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-    openexr ENABLE_OPENEXR_SUPPORT
+    FEATURES openexr ENABLE_OPENEXR_SUPPORT
+    INVERTED_FEATURES win7 BUILD_DX12
```
I thought DirectX 12 was backported to Windows 7 by Microsoft? Regardless, It seems awkward to require opting-into a feature to get Windows 7 support. I realize that Windows 7 is no longer supported by Microsoft - but my personal opinion is that it would make more sense to have a feature for opting-in to DX12.

RT222 replied:
I'm not entirely sure how the DX12 on Windows 7 works, but DirectXTex only enable it for Windows 10 as explained here : https://github.com/microsoft/DirectXTex/wiki/DirectXTex#using-project-to-project-references

I think you're right about the feature, and the previous port was building without DX12 support anyway so there's less chances to break things for users if we keep it that way. I'm changing it.

RT2Code committed:
[directxtex] Replace win7 with dx12 feature

RT2Code committed:
[directxrex] Shorten patch file

RT2Code committed:
[directxtex] Fix wrong port version

RT2Code committed:
[directxtex] Use ninja
